lib/transport.ts

```diff
@@ -26,14 +26,25 @@ export class Transport {
   site: string;
   apiKey?: string;
   apiKMSKey?: string;
+  extensionLayerVersion?: number;
 
-  constructor(flushMetricsToLogs?: boolean, site?: string, apiKey?: string, apiKMSKey?: string) {
+  constructor(
+    flushMetricsToLogs?: boolean,
+    site?: string,
+    apiKey?: string,
+    apiKMSKey?: string,
+    extensionLayerVersion?: number,
+  ) {
     if (flushMetricsToLogs === undefined) {
       this.flushMetricsToLogs = transportDefaults.flushMetricsToLogs;
     } else {
       this.flushMetricsToLogs = flushMetricsToLogs;
     }
 
+    if (extensionLayerVersion !== undefined) {
+      this.flushMetricsToLogs = false;
+    }
```

Comment:
If the customer requests to install the extension layer ^ then we override flushMetricsToLogs to be false so that the apiKey will be read and added as an environment variable. Addresses our slack convo

Reply:
I would add a short comment in the code explaining why we set flushMetricsToLogs as false: if the extension is used, metrics will be submitted via the extension.

Also, can we just make it so that we ALWAYS set the API key on the Lambda functions if an API key is provided? There is no downside I am aware of in setting that environment variable, so even if flush metrics to logs is enabled and it's not strictly necessary, it seems that going out of our way to NOT set the API key if we have one just makes things more complicated.

Reply:
changes in setEnvVars() in transport.ts file should address the comment above ^